diff --git a/proc/slab.c b/proc/slab.c
index 60c5696..bfa74f1 100644
--- a/proc/slab.c
+++ b/proc/slab.c
@@ -166,7 +166,7 @@ static int parse_slabinfo20(struct slab_info **list, struct slab_stat *stats,
 		curr->cache_size = (unsigned long)curr->nr_slabs * curr->pages_per_slab * page_size;
 
 		if (curr->nr_objs) {
-			curr->use = 100 * curr->nr_active_objs / curr->nr_objs;
+			curr->use = (unsigned)(100 * (float)curr->nr_active_objs / curr->nr_objs);
 			stats->nr_active_caches++;
 		} else
 			curr->use = 0;
@@ -244,7 +244,7 @@ static int parse_slabinfo11(struct slab_info **list, struct slab_stat *stats,
 		curr->cache_size = (unsigned long)curr->nr_slabs * curr->pages_per_slab * page_size;
 
 		if (curr->nr_objs) {
-			curr->use = 100 * curr->nr_active_objs / curr->nr_objs;
+			curr->use = (unsigned)(100 * (float)curr->nr_active_objs / curr->nr_objs);
 			stats->nr_active_caches++;
 		} else
 			curr->use = 0;
