diff -Naur procps-3.2.8.orig/sysctl.c procps-3.2.8/sysctl.c
--- procps-3.2.8.orig/sysctl.c	2015-12-16 12:35:07.000000000 +0100
+++ procps-3.2.8/sysctl.c	2015-12-16 12:37:49.386001332 +0100
@@ -421,7 +421,7 @@
    return rc;
 }
 
-
+#define LINELEN 4096
 
 /*
  *     Preload the sysctl's from the conf file
@@ -429,8 +429,8 @@
  *
  */
 static int Preload(const char *restrict const filename) {
-   char oneline[256];
-   char buffer[256];
+   char oneline[LINELEN];
+   char buffer[LINELEN];
    FILE *fp;
    char *t;
    int n = 0;
