diff --git a/top.c b/top.c
index 53e290f..2230d7d 100644
--- a/top.c
+++ b/top.c
@@ -1150,6 +1150,10 @@ static proc_t **procs_refresh (proc_t **table, int flags)
    else
       PT = openproc(flags);
 
+   if (!PT) {
+      std_err("/proc is not mounted, required for output data");
+      return table;
+   }
    // i) Allocated Chunks:  *Existing* table;  refresh + reuse
    if (!(CHKw(Curwin, Show_THREADS))) {
       while (curmax < savmax) {
