diff -up ./ps/global.c.ori ./ps/global.c
--- ./ps/global.c.ori	2017-06-07 10:36:41.967984768 +0200
+++ ./ps/global.c	2017-06-07 10:38:53.574255944 +0200
@@ -362,21 +362,22 @@ void reset_global(void){
   look_up_our_self(&p);
   set_screen_size();
   set_personality();
-  int fd;
+  FILE *fd;
   char *buf[BUFFSIZE];
   const char *b;
 
   /* get boot time from /proc/stat */
-  fd = open("/proc/stat", O_RDONLY, 0);
-  if (fd != -1) {
-    buf[BUFFSIZE-1] = 0;
-    read(fd, buf, BUFFSIZE-1);
-    b = strstr(buf, "btime ");
-    if (b) {
-      sscanf(b, "btime %lu", &time_of_boot);
-      seconds_since_boot = time(0) - time_of_boot;
-    }  
-    close(fd);
+  fd = fopen("/proc/stat", "r");
+  if (fd) {
+    while (fgets(buf, sizeof(buf), fd)) {
+      b = strstr(buf, "btime ");
+      if (b) {
+        sscanf(b, "btime %lu", &time_of_boot);
+        seconds_since_boot = time(0) - time_of_boot;
+        break;
+      }
+    }
+    fclose(fd);
   }
     
   all_processes         = 0;
