diff -Naur procps-3.2.8.orig/pmap.c procps-3.2.8/pmap.c
--- procps-3.2.8.orig/pmap.c	2015-12-04 17:08:19.000000000 +0100
+++ procps-3.2.8/pmap.c	2015-12-04 17:04:42.000000000 +0100
@@ -289,10 +289,12 @@
 
         );
       }else{
-        printf("-------- ------- ------- ------- -------\n");
+        printf("--------  ------  ------  ------\n");
         printf(
-          "total kB %7ld       -       -       -\n",
-          (total_shared + total_private_writeable + total_private_readonly) >> 10
+          "total kB %7ld %7llu %7llu\n",
+          (total_shared + total_private_writeable + total_private_readonly) >> 10,
+          total_rss, (total_shared_dirty+total_private_dirty)
+
         );
       }
     }
