diff -Naur procps-3.2.8.orig/pmap.c procps-3.2.8/pmap.c
--- procps-3.2.8.orig/pmap.c	2015-11-03 14:25:10.000000000 +0100
+++ procps-3.2.8/pmap.c	2015-11-03 14:29:07.699010454 +0100
@@ -209,9 +209,9 @@
           rss = shared_dirty = private_dirty = 0ull;
           continue;
         }
-        /* Other keys */
-        continue;
       }
+      /* Anything else starting with A-Z? -> skip it (rhbz#1262870) */
+      continue;
     }
     sscanf(mapbuf,"%"KLF"x-%"KLF"x %31s %Lx %x:%x %Lu", &start, &end, flags, &file_offset, &dev_major, &dev_minor, &inode);
 
